keps/sig-network/20190603-EndpointSlice-API.md

```diff
@@ -59,81 +59,116 @@ type EndpointSlice struct {
     metav1.TypeMeta `json:",inline"`
     // OwnerReferences should be set when the object is derived from a k8s
     // object.
-    // The object labels may contain the following key:
+    // The object labels may contain the following keys:
     // kubernetes.io/service: the label value indicates the name of the
     // service from which the EndpointSlice is derived from.
+    // endpointslice.kubernetes.io/managed-by: the label value represents a
```

Comment:
We'll need a bit more spec on what value that should take.  Labels are pretty limited in what they can contain as values.

Another approach might be to add a per-owner label so the key becomes significant.  E.g. `endpointslice.kubernetes.io/controller-managed: "true"`

This allows multiple controllers to think they own a slice, but I am not sure that will be a problem in practice.

Reply:
Yeah it's hard to come up with the right label(s) here. Would an EndpointSlice ever become managed by a different resource? I think that's unlikely, but if it happened it would involve controllers changing labels that weren't managed by them (or like you mentioned, just letting multiple controllers try to manage a slice). 

If we were to switch the important information from the value to the label (`endpointslice.kubernetes.io/controller-managed: "true"`), what would other related labels look like? As an example, I think `kube-apiserver` would be a value for `endpointslice.kubernetes.io/managed-by` for the initial EndpointSlice created/managed by the apiserver, would we want a separate label for that?

Assuming this is all to support additional controllers managing EndpointSlices, I think the `controller-managed` name might become confusing as it might seem to indicate the difference between an EndpointSlice being managed by _any_ controller vs being created manually.

Reply:
If I had a 3rd party controller it might be `example.com/tims-cool-controller: "true"`

re apiserver: whether we use label keys or values, it seems appropriate to denote that the apiserver controls its own endpoints.

By now you should know not to let me name things.  `endpointslice.kubernetes.io/managed-by-endpointslice-controller: "true"` is just as fine.  No humans will type this.

My main problem with the simpler proposal is that nothing prevents me from accidentally writing a controller called "cool-controller" and you from doing the exact same.